os-server/src/main/java/app/openschool/discussion/DiscussionQuestionMentorServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+package app.openschool.discussion;
+
+import app.openschool.common.exceptionhandler.exception.PermissionDeniedException;
+import app.openschool.course.Course;
+import app.openschool.course.CourseRepository;
+import app.openschool.course.EnrolledCourse;
+import app.openschool.course.EnrolledCourseRepository;
+import app.openschool.discussion.dto.DiscussionQuestionRequestDto;
+import app.openschool.discussion.dto.DiscussionQuestionResponseDto;
+import app.openschool.discussion.mapper.DiscussionQuestionMentorMapper;
+import app.openschool.user.User;
+import app.openschool.user.UserRepository;
+import java.security.Principal;
+import java.time.Instant;
+import org.springframework.stereotype.Service;
+
+
+@Service("discussionQuestionMentor")
+public class DiscussionQuestionMentorServiceImpl implements DiscussionQuestionService {
+
+  private final UserRepository userRepository;
+  private final EnrolledCourseRepository enrolledCourseRepository;
+  private final CourseRepository courseRepository;
+  private final DiscussionQuestionMentorRepository discussionQuestionMentorRepository;
+
+  public DiscussionQuestionMentorServiceImpl(
+      UserRepository userRepository,
+      EnrolledCourseRepository enrolledCourseRepository,
+      CourseRepository courseRepository,
+      DiscussionQuestionMentorRepository discussionQuestionMentorRepository) {
+    this.userRepository = userRepository;
+    this.enrolledCourseRepository = enrolledCourseRepository;
+    this.courseRepository = courseRepository;
+    this.discussionQuestionMentorRepository = discussionQuestionMentorRepository;
+  }
+
+  @Override
+  public DiscussionQuestionResponseDto create(
+      DiscussionQuestionRequestDto requestDto, Principal principal) {
+    User user = userRepository.findUserByEmail(principal.getName());
+    EnrolledCourse enrolledCourse =
+        enrolledCourseRepository
+            .findByCourseId(requestDto.getCourseId())
+            .orElseThrow(IllegalArgumentException::new);
+    if (enrolledCourse.getUser().getId() != user.getId()) {
+      throw new PermissionDeniedException("permission.denied");
+    }
+
+    return DiscussionQuestionMentorMapper.toResponseDto(
+        discussionQuestionMentorRepository.save(creteQuestion(requestDto, principal)));
+  }
+
+  private DiscussionQuestionMentor creteQuestion(
+      DiscussionQuestionRequestDto requestDto, Principal principal) {
```

Comment:
please do not pass the principal inside the service if you need only the username just get it and pass it to the service, not whole principal